# Introduction

The directory is based on South Korea's biggest lottery **Lotto 6/45**.

## Setup

In [29]:
from lib.nCr import nCr

**Let us calculate the probabilities of each price:**

In [30]:
total = nCr(45, 6)

# Getting all 6 numbers right
prob = total / nCr(6, 6) 
print(f'1st Place Probability : 1 / {prob:,}')

# Getting 5 numbers, bonus right, selecting 0 wrong number
prob = total / (nCr(6, 5) * nCr(39, 0) * nCr(1, 1)) 
print(f'2nd Place Probability : 1 / {prob:,}')

# Getting 5 numbers, selecting 1 wrong number, 
# Excluding 6 of 2nd place possible cases
prob = total / (nCr(6, 5) * nCr(39, 1) - 6)
print(f'3rd Place Probability : 1 / {prob:,}')

# Getting 4 numbers, selecting 2 wrong numbers
prob = total / (nCr(6, 4) * nCr(39, 2))
print(f'4th Place Probability : 1 / {prob:,}')

# Getting 3 numbers, selecting 3 wrong numbers
prob = total / (nCr(6, 3) * nCr(39, 3))
print(f'5th Place Probability : 1 / {prob:,}')

1st Place Probability : 1 / 8,145,060.0
2nd Place Probability : 1 / 1,357,510.0
3rd Place Probability : 1 / 35,723.94736842105
4th Place Probability : 1 / 732.7989203778677
5th Place Probability : 1 / 44.56209650946493


**P (Total Price Money)= 0.5 * (Total Lottery Sold)**<br>**A = P - (Money Distributed to 4th & 5th prizes)**
| Rank | Price | Probability | Note |
| :-: | :-: | --: | :-: |
| 1 | 75% of A(Total) | 1 / 8,145,060 | 6 Numbers Match |
| 2 | 12.5% of A(Total) | 1 / 1,357,510 | 5 Numbers + Bonus Number Match |
| 3 | 12.5% of A(Total) | ≈ 1 / 35,723 | 5 Numbers Match |
| 4 | ₩50,000 | ≈ 1 / 733 | 4 Numbers Match |
| 5 | ₩5,000 | ≈ 1 / 45 | 3 Numbers Match |